internal/logging/logging.go

```diff
@@ -0,0 +1,185 @@
+package logging
+
+import (
+	"fmt"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"os"
+	"sync"
+	"time"
+)
+
+// Logging implements logging using: a switch to disable logging on Fatal log, default logger and level,
+// and stored child loggers and their options.
+type Logging struct {
+	// enabled implements a switch to disable logging on Fatal log.
+	enabled *atomic.Bool
+	// level defines default log-level.
+	level zap.AtomicLevel
+	// output defines default log-level.
+	output string
+	// logger defines default logger.
+	logger *zap.SugaredLogger
+	// encoder stores zapcore.Encoder; used to build the logger
+	encoder zapcore.Encoder
+	// encoder stores zapcore.WriteSyncer; used to build the logger
```

Comment:
// syncer defines the zapcore.WriterSyncer,
// which is used to create the default logger and the child loggers